ScienceDirect.js

```diff
@@ -209,31 +211,36 @@ function processRIS(doc, text) {
 			if (name.indexOf(',') == -1) {
 				name = name.trim().replace(/^(.+?)\s+(\S+)$/, '$2, $1');
 			}
-			
+
 			return pre + name;
 		}
 	);
+	//Z.debug(text)
+	//The RIS sometimes has spaces at the beginning of lines, which break things
+	//as of 20170121 e.g. on http://www.sciencedirect.com/science/article/pii/B9780123706263000508 for A2
+	//remove them
+	text = text.replace(/\n\s+([A-Z][A-Z0-9])\s{2}\-\s/, "\n$1  - ");
```

Comment:
How about simplifying this with
```js
text = text.replace(/\n\s*/, "\n");
```
? Because the `*` is greedy, it should delete all spaces in the beginning of lines, and it seems less fragile than your regexp.